planner/core/rule_aggregation_elimination.go

```diff
@@ -69,6 +69,39 @@ func (a *aggregationEliminateChecker) tryToEliminateAggregation(agg *LogicalAggr
 	return nil
 }
 
+func (a *aggregationEliminateChecker) tryToEliminateDistinct(agg *LogicalAggregation) {
+	if len(agg.GetGroupByCols()) != 0 {
+		return
+	}
+	for _, af := range agg.AggFuncs {
+		if af.HasDistinct {
+			cols := make([]*expression.Column, 0, len(af.Args))
+			canEliminate := true
+			for _, arg := range af.Args {
+				if col, ok := arg.(*expression.Column); ok {
+					cols = append(cols, col)
+				} else {
+					canEliminate = false
+					break
+				}
+			}
+			if canEliminate {
+				distinctByUniqueKey := false
+				schemaByDistinct := expression.NewSchema(cols...)
+				for _, key := range agg.children[0].Schema().Keys {
```

Comment:
Shall we check `Schema.UniqueKeys` as well? AFAICS, the result of common aggregations on columns would not be impacted by the `null` values in the column. e.g,

```sql
create table t(a int, unique key(a));
explain select count(distinct) a from t;
```

@winoros can you think of any aggregation case that would break this assumption?

Reply:
Oh yeah! I forgot that one :joy:

Reply:
This check has been added.